File: 28702/28702.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    28702.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/07/21 18:06:50 by jihoolee          #+#    #+#              #
#    Updated: 2024/07/21 19:39:02 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

WORD_CNT: int = 3


def fizzbuzz(num: int):
    if num % 3 == 0 and num % 5 == 0:
        return 'FizzBuzz'
    elif num % 3 == 0:
        return 'Fizz'
    elif num % 5 == 0:
        return 'Buzz'
    else:
        return num


def main() -> None:
    num_strs: list[str] = [input() for _ in range(WORD_CNT)]

    for idx in range(len(num_strs)):
        if num_strs[idx] not in ['Fizz', 'Buzz', 'FizzBuzz']:
            num: int = int(num_strs[idx]) + WORD_CNT - idx
    print(fizzbuzz(num))


if __name__ == '__main__':
    main()

File: 28702/28702.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   28702.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/07/21 17:29:52 by jihoolee          #+#    #+#             */
/*   Updated: 2024/07/21 18:01:00 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>

const size_t WORD_CNT = 3;

size_t get_numeric_elem_idx(const std::vector<std::string>& v) {
  size_t idx = 0;

  for (idx = 0; idx < v.size(); ++idx) {
    std::string substring = v[idx].substr(0, 4);

    if (substring != "Fizz" && substring != "Buzz")
      break;
  }
  return idx;
}

void print_fizzbuzz(int num) {
  if (num % 3 == 0 && num % 5 == 0)
    std::cout << "FizzBuzz";
  else if (num % 3 == 0)
    std::cout << "Fizz";
  else if (num % 5 == 0)
    std::cout << "Buzz";
  else
    std::cout << num;
}

int main(void) {
  std::vector<std::string> v(WORD_CNT);

  for (size_t idx = 0; idx < WORD_CNT; ++idx)
    std::cin >> v[idx];

  size_t  numeric_idx = get_numeric_elem_idx(v);

  print_fizzbuzz(std::stoi(v[numeric_idx]) + WORD_CNT - numeric_idx);
  return 0;
}
